test/dataloader/ecto/multi_sync_test.exs

```diff
@@ -0,0 +1,49 @@
+defmodule Dataloader.Ecto.MultiSyncTest do
+  use ExUnit.Case, async: false
+
+  alias Dataloader.User
+  alias Dataloader.TestRepo, as: Repo
+
+  setup do
+    :ok = Ecto.Adapters.SQL.Sandbox.checkout(Repo)
+    :ok = Ecto.Adapters.SQL.Sandbox.mode(Repo, {:shared, self()})
+
+    test_pid = self()
+
+    source =
+      Dataloader.Ecto.new(
+        Repo,
+        query: &query(&1, &2, test_pid)
+      )
+
+    loader =
+      Dataloader.new()
+      |> Dataloader.add_source(Test, source)
+
+    {:ok, loader: loader}
+  end
+
+  defp query(schema, _params, test_pid) do
+    send(test_pid, :querying)
+
+    schema
+  end
+
+  test "loads data", %{loader: loader} do
+    user =
+      %User{username: "Ben Wilson"}
+      |> Repo.insert!()
+
+    result =
+      Ecto.Multi.new()
+      |> Ecto.Multi.run(:loader, fn _, _ ->
+        loader
+        |> Dataloader.load(Test, User, user.id)
+        |> Dataloader.run()
```

Comment:
All of the errors actually show that the error is that this isn't returning an `:ok` tuple. If you fix that issue do the other issues persist?

Reply:
Oh, you're absolutely right, I misinterpreted the error message. It does work when returning an `:ok` tuple 😅 I'm gonna close the PR and issue then. Thanks a lot for your time and help! 🙏